Cleanup fml package and fix HoverChecker

ichttt opened the pull request:
Removes the old config gui stuff, and patches HoverChecker to be more generic in preparation for #6421
The gui classes where repackaged and renamed to match the new mcp naming and move them out the config package (as it is not really appropriate).
Also removes the following classes that are not needed anymore (since new fml):
- GuiCheckBox (Vanilla has a proper CheckBox now)
- ResourceLocationUtils (Does not provide much value to mods)
- StackTraceUtils (Not much value for mods)
- ClassNameUtils (Basically Class.getSimpleName)
- IFMLHandledException
- IWorldGenerator
- IGuiHandler
- ExistingSubstitutionException (Even unused in 1.12)
- TextureTracker (now in ClientHooks)
- GuiAccessDenied (Blocking fml clients seems to be gone in 1.13+, and I don't think it was even used in 1.12 by anyone)
- GuiBackupFailed (Vanilla has it's own backup system now)

ModFileResourcePack was updated as it contained imports to a now deleted class, which is not used in the class.

ichttt committed:
Cleanup client package to remove unused stuff

pupnewfster commented on src/main/java/net/minecraftforge/fml/client/config/HoverChecker.java:
```diff
@@ -19,6 +19,7 @@
 
 package net.minecraftforge.fml.client.config;
 
+import net.minecraft.client.gui.widget.Widget;
```
Can the other import for button be removed now? Glancing at the class it seems like it could be but didn't open up a workspace to verify

raoulvdberge commented:
Can you remove IGuiHandler too?

ichttt committed:
Also remove legacy classes from fml.common and fml root package

Cadiboo commented:
Since this is already a breaking change, I think it would be good to also move fml’s GUI util classes like GuiButtonExt and HoverChecker to their own GUI utils package & rename them to be consistent with the new suffix based naming.

ichttt committed:
Refactor: Rename and repackage fml gui classes

Cadiboo commented on src/main/java/net/minecraftforge/fml/client/gui/screen/ModListScreen.java:
```diff
@@ -123,7 +123,7 @@ public ModListScreen(Screen mainMenu)
         super(new TranslationTextComponent("fml.menu.mods.title"));
         this.mainMenu = mainMenu;
-        this.mods = Collections.unmodifiableList(ModList.get().getMods());
+        this.mods = Collections.unmodifiableList(net.minecraftforge.fml.ModList.get().getMods());
```
Couldn’t this be an import instead of a fully qualified reference?

TheLimePixel replied:
https://github.com/MinecraftForge/MinecraftForge/wiki/If-you-want-to-contribute-to-Forge#conventions-for-coding-patches-for-a-minecraft-class-javapatch

LexManos replied:
This isn't a patch file.

Cadiboo commented on src/main/java/net/minecraftforge/fml/client/gui/screen/ModListScreen.java:
```diff
@@ -450,7 +450,7 @@ public void updateDynamicTexture() {
         lines.add(selectedMod.getDisplayName());
         lines.add(ForgeI18n.parseMessage("fml.menu.mods.info.version", MavenVersionStringHelper.artifactVersionToString(selectedMod.getVersion())));
-        lines.add(ForgeI18n.parseMessage("fml.menu.mods.info.idstate", selectedMod.getModId(), ModList.get().getModContainerById(selectedMod.getModId()).
+        lines.add(ForgeI18n.parseMessage("fml.menu.mods.info.idstate", selectedMod.getModId(), net.minecraftforge.fml.ModList.get().getModContainerById(selectedMod.getModId()).
```
Couldn’t this be an import instead of a qualified reference?

ichttt replied:
Because I renamed the ModListWidget to just ModList, which is now I duplicate name. I will revert it to ModListWidget to avoid confusion

LexManos replied:
And this is a indication that you shouldn't of named it that.

ichttt committed:
Fix ModListWidget name to avoid confusion and collision.
Also delete CheckBox, as vanilla has it's own CheckBox now which looks much better

ichttt commented:
Should be fine now. I also removed the fml GuiCheckBox, as vanilla now has it's own CheckBox and fml's was just a hijacked button anyway

Cadiboo commented:
Could this also be extended to include [these changes](https://github.com/MinecraftForge/MinecraftForge/pull/6421/commits/af46c3a6cd15a55d59450dcf4f04bd15714b549e)?

ichttt commented:
@Cadiboo I don't know what you mean, that link goes nowhere

Cadiboo commented:
My bad, I meant The commit that updates the GuiMessageDialog https://github.com/MinecraftForge/MinecraftForge/pull/6421/commits/ed7ba8330b9de26b2020cc4a5b524ce00b16296f. I rebased and forgot to update the link

ichttt commented:
I already remove the old class in this PR, I think the new code should stay in your PR as I just want this PR to be easily reviewed and only consist of breaking changes.

Cadiboo commented:
Sounds good. Should I now base my PR on top of yours and make a separate change for all the other breaking changes in my PR?

Cadiboo committed:
Rename ModConfigEvent.ConfigReloading to ModConfigEvent.Reloading

Cadiboo commented:
I've made [my own branch in my repo](https://github.com/Cadiboo/MinecraftForge/tree/config-breaking-changes) for a couple other breaking changes that I'd like you to include in your PR.
Those are the only other breaking changes I need added for my config screen PR.

ichttt committed:
Merge branch '1.15.x' of https://github.com/MinecraftForge/MinecraftForge into configGuiPreparations

ichttt committed:
Merge branch '1.15.x' of https://github.com/MinecraftForge/MinecraftForge into configGuiPreparations
